### PR TITLE
CPO-17 Create initial Functional Test library

### PR DESCRIPTION
**Pull request by mattnayler**
### JIRA link (if applicable) ###

   [CPO-17](https://tools.hmcts.net/jira/browse/CPO-17) _"Create initial Functional / Smoke tests library for CPO"_

### Change description ###

Create initial Functional / Smoke tests library for CPO.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by sullivangareth — APPROVED** [review]

File: Jenkinsfile_CNP [hunk 1]
```diff
@@ -10,10 +10,97 @@ def component = "case-payment-orders-api"
...
+static LinkedHashMap<String, Object> secret(String secretName, String envVar) {
+  [$class     : 'AzureKeyVaultSecret',
+    secretType : 'Secret',
+    name       : secretName,
+    version    : '',
+    envVariable: envVar
+  ]
+}
+
+def vaultOverrides = [
```
Comment (smathangi) [inline Jenkinsfile_CNP:37]: i think this is not required any more. we have to cleanup in other apps soon.

File: README.md [hunk 2]
```diff
@@ -89,6 +89,68 @@ docker image rm <image-id>
...
+
+> Note: These are the tests run against an environment.  This environment must include access to
+> IDAM and Service Auth Provider services.  All can be made available by installing and running the
+> CCD stack as advised in [ccd-docker](https://github.com/hmcts/ccd-docker), with the minimum
+> enabled services `sidam sidam-local sidam-local-ccd`.
+>
+> If you would like to test your local environment you'll need to export variables on your
+> `.bash_profile` script.  The [ccd-docker](https://github.com/hmcts/ccd-docker) repository
+> contains its own scripts to set the required environment variables.  Otherwise, the details of
+> the required minimum set of variables can be found by reviewing the contents of this project's
```
Comment (smathangi) [inline README.md:125]: ⭐

**Review by smathangi — APPROVED** [review]
LGTM 👍

File: Jenkinsfile_CNP [hunk 3]
```diff
@@ -10,10 +10,97 @@ def component = "case-payment-orders-api"
...
-    }
+    builder.gradle('integration')
+
+    // hmcts/cnp-jenkins-library may fail to copy artifacts after checkstyle error so repeat command (see /src/uk/gov/hmcts/contino/GradleBuilder.groovy)
+    steps.archiveArtifacts allowEmptyArchive: true, artifacts: '**/reports/checkstyle/*.html'
+
+    steps.archiveArtifacts allowEmptyArchive: true, artifacts: '**/reports/pmd/*.html'
+  }
+
+  overrideVaultEnvironments(vaultOverrides)
```
Comment (smathangi) [inline Jenkinsfile_CNP:83]: same as above

File: Jenkinsfile_CNP (hunk 1 above)
Comment (mattnayler) [reply Jenkinsfile_CNP]: Thanks, removed in 724fb6289123e7a62d84aa0bb7ac3181934cf454.

File: Jenkinsfile_CNP (hunk 3 above)
Comment (mattnayler) [reply Jenkinsfile_CNP]: Thanks, removed in 724fb6289123e7a62d84aa0bb7ac3181934cf454.

**Pull request merged**